SignalServiceKit/src/Util/OWSAnalytics.h

```diff
@@ -115,14 +117,33 @@ typedef NSDictionary<NSString *, id> *_Nonnull (^OWSProdAssertParametersBlock)()
 #define AnalyticsParametersFromNSError(__nserror)                                                                      \
     ^{                                                                                                                 \
         return (@{                                                                                                     \
-            kOWSProdAssertParameterNSErrorDomain : __nserror.domain,                                                   \
+            kOWSProdAssertParameterNSErrorDomain : (__nserror.domain ?: @"unknown"),                                   \
             kOWSProdAssertParameterNSErrorCode : @(__nserror.code),                                                    \
-            kOWSProdAssertParameterNSErrorDescription : __nserror.description,                                         \
+            kOWSProdAssertParameterNSErrorDescription : (__nserror.description ?: @"unknown"),                         \
+        });                                                                                                            \
+    }
+
+#define AnalyticsParametersFromNSException(__exception)                                                                \
+    ^{                                                                                                                 \
+        return (@{                                                                                                     \
+            kOWSProdAssertParameterNSExceptionName : (__exception.name ?: @"unknown"),                                 \
+            kOWSProdAssertParameterNSExceptionReason : (__exception.reason ?: @"unknown"),                             \
+            kOWSProdAssertParameterNSExceptionClassName :                                                              \
+                (__exception ? NSStringFromClass([__exception class]) : @"unknown"),                                   \
         });                                                                                                            \
     }
 
 #define OWSProdFailWNSError(__analyticsEventName, __nserror)                                                           \
-    OWSProdFailWParams(__analyticsEventName, AnalyticsParametersFromNSError(__nserror))
+    {                                                                                                                  \
+        DDLogError(@"%s:%d %@: %@", __PRETTY_FUNCTION__, __LINE__, __analyticsEventName, __nserror.debugDescription);  \
```

Comment:
We want the debug logs to be more verbose than the analytics events.  In this case the error's `debugDescription` is considered valuable enough to log but too dangerous to include in the analytics event.  I'll add a comment.